### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by Arthur-Antonyan**

### CODE REVIEW

File: sudoku/index.html [hunk 1]
```diff
@@ -3,13 +3,13 @@
   <head>
     <meta charset="UTF-8" />
     <meta http-equiv="X-UA-Compatible" content="IE=edge" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <metat" content="width=device-width, initial-scale=1.0" />
```
Comment (lilithkarapetyan) [inline sudoku/index.html:6]: is this by accident?

File: sudoku/script.js [hunk 2]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+      checkColumnsArr.push(matrix[j][i]);
+      z = matrix[j].length - 1;
+    }
+    let x = new Set(checkColumnsArr);
+    let y = temp;
+    for (let props of x) {
+      if (props === y || x.size != z) {
+        answers.push(false);
+        break;
+      } else continue;
```
Comment (lilithkarapetyan) [inline sudoku/script.js]:
```suggestion
      }
```

File: sudoku/script.js [hunk 3]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+      checkRowsArr.push(matrix[i][j]);
+      z = matrix[i].length - 1;
+    }
+    let x = new Set(checkRowsArr);
+    let y = temp;
+    for (let props of x) {
+      if (props === y || x.size != z) {
+        answers.push(false);
+        break;
+      } else continue;
```
Comment (lilithkarapetyan) [inline sudoku/script.js]: No need for this, anyways you don't have anything after continue
```suggestion
      } 
```

File: sudoku/script.js [hunk 4]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+
+let checkColumnsArr = [];
+function checkColumns(matrix) {
+  let temp, z;
+  for (let i = 0; i < matrix.length; i++) {
+    checkColumnsArr.splice(0, z);
+    for (let j = 1; j < matrix[i].length; j++) {
+      temp = matrix[0][i];
+      checkColumnsArr.push(matrix[j][i]);
+      z = matrix[j].length - 1;
```
Comment (lilithkarapetyan) [inline sudoku/script.js:64]: Couldn't we combine the checkColumns and checkRows? Mostly they are the same. Depending on a boolean value you can write an `if` for this part and reuse the code

File: sudoku/script.js [hunk 5]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+      } else continue;
+    }
+  }
+}
+
+let matrix = [];
+let arr = [];
+
+function split(arr, len) {
+  (i = 0), (n = arr.length);
```
Comment (lilithkarapetyan) [inline sudoku/script.js:102]: are those global variables?

File: sudoku/script.js [hunk 6]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+  while (i < n) {
+    matrix.push(arr.slice(i, (i += len)));
+  }
+  return matrix;
+}
+
+check.addEventListener('click', () => {
```
Comment (lilithkarapetyan) [inline sudoku/script.js:109]: the file is too big, you can move the event listeners to a separate file

File: sudoku/script.js [hunk 7]
```diff
@@ -1,8 +1,21 @@
...
+let div = document.querySelectorAll('.board>div');
 let btn = document.querySelector('.change');
+let check = document.querySelector('.check');
+let answers = [];
 for (let i = 0; i < div.length; i++) {
   div[i].classList.add(i + 1);
 }
+let wrapper = document.querySelector('.board');
+wrapper.oninput = (event) => {
+  let regex = /^\d$/;
```
Comment (robhov97) [inline sudoku/script.js]: Seems like you didn't changed this variable, you can specify it as a constant
```suggestion
  const regex = /^\d$/;
```

File: sudoku/script.js [hunk 8]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+function checkColumns(matrix) {
+  let temp, z;
+  for (let i = 0; i < matrix.length; i++) {
+    checkColumnsArr.splice(0, z);
+    for (let j = 1; j < matrix[i].length; j++) {
+      temp = matrix[0][i];
+      checkColumnsArr.push(matrix[j][i]);
+      z = matrix[j].length - 1;
+    }
+    let x = new Set(checkColumnsArr);
```
Comment (robhov97) [inline sudoku/script.js]:
```suggestion
    const x = new Set(checkColumnsArr);
```

File: sudoku/script.js [hunk 9]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
...
+    for (let props of x) {
+      if (props === y || x.size != z) {
+        answers.push(false);
+        break;
+      } else continue;
+    }
+  }
+}
+
+let checkRowsArr = [];
```
Comment (robhov97) [inline sudoku/script.js]:
```suggestion
const checkRowsArr = [];
```

File: sudoku/script.js [hunk 10]
```diff
@@ -39,3 +52,92 @@ btn.addEventListener('click', () => {
   }
 });
 btn.addEventListener('click', addRandom);
+
+let checkColumnsArr = [];
```
Comment (robhov97) [inline sudoku/script.js]:
```suggestion
const checkColumnsArr = [];
```

**Review by robhov97 — CHANGES_REQUESTED** [review]

**Comment by robhov97** [comment]
`.gitignore` is empty. For now you can put `.idea` and `.vscode` in it

File: sudoku/index.html (hunk 1 above)
Comment (Arthur-Antonyan) [reply sudoku/index.html:6]: yep. deleted

File: sudoku/script.js (hunk 5 above)
Comment (Arthur-Antonyan) [reply sudoku/script.js:102]: nope. corrected

File: sudoku/script.js (hunk 4 above)
Comment (Arthur-Antonyan) [reply sudoku/script.js:64]: sure, will optimize later

**Review by robhov97 — APPROVED** [review]

**Pull request merged**